Require wire_name's to be unique

nsanch opened the pull request:
Update ThriftValidator to require that no two fields in a struct share
the same wire_name (and that you can't have wire_name X if there's a
field X with no explicit wire_name).

It's possible this belongs in ScalaClass, not ThriftValidator. Let me know if so.

Commit:
Require wire_name's to be unique

Update ThriftValidator to require that no two fields in a struct share
the same wire_name (and that you can't have wire_name X if there's a
field X with no explicit wire_name).

jorgeortiz85 commented:
Yeah, can we put this in ScalaClass? There's already some validation of wire_names there (for retired_wire_names). Also, I'd like for ThriftParser to be usable with any Thrift code, and perhaps someone else has a different meaning for a "wire_name" annotation than we do.

nsanch commented:
yep, will move it.

Commit:
move repeated wire_name validation to ScalaClass

hard bits are for testing:

- make ThriftCodegenTest that calls into ThriftCodegen to get a
  ScalaProgram
- expose a method on ThriftCodegen to make that test easier.
- make a new directory and sbt sub-project for this test
